criu/fdstore.c

```diff
@@ -49,8 +53,16 @@ int fdstore_init(void)
 		return -1;
 	}
 
-	if (setsockopt(sk, SOL_SOCKET, SO_SNDBUFFORCE, &buf[0], sizeof(buf[0])) < 0 ||
```

Comment:
we can try to run SO_SNDBUFFORCE and than fall back to SO_RCVBUF if opts.unprivileged is set.

Reply:
That would be simpler and nicer I agree. On the other hand it would require 4 syscalls total for non-root non-CAP_NET_ADMIN processes, v.s. 2 currently. Since this code is on the restore path can we keep the "free" bit of extra performance we have with the current patch?

(Also I assume you meant SO_SNDBUFFORCE fall back to SO_SNDBUF and SO_RCVBUFFORCE fall back to SO_RCVBUF.)

Reply:
ack-ed.

Reply:
FYI, the kernel uses `sockopt_capable(CAP_NET_ADMIN)` rather than `sockopt_ns_capable(CAP_NET_ADMIN)` (e.g. see code `sk_setsockopt` on HEAD) to check permissions. This means that for a namespaced process with CAP_NET_ADMIN inside the namespace but not outside, this code would attempt to use *FORCE and the kernel would reject it.

Reply:
You're right. @avagin's original suggestion of how this should be done would have handled that more gracefully. I'll open a PR.